### DISCUSSION TITLE
Welcome to fingerprintjs Discussions!

### DISCUSSION THREAD

**Discussion by Valve** [fingerprintjs/fingerprintjs#620] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Chaitanyarajuk** [comment]
What is the difference between Master branch(3.1.2) and v2? 
Master means Pro/paid version?

**Reply by Finesse** [reply]
Both of the branches are OSS/free. The current `master` is version 3 and `v2` is version 2 of the same project. Version 3 is evolution of version 2, see what was changed in version 3 in [the release notes](https://github.com/fingerprintjs/fingerprintjs/releases/tag/v3.0.0). Version 3 continues the project improvement while version 2 gets only bug fixes.

**Reply by Chaitanyarajuk** [reply]
OK, Thank you Finesse for the quick response

**Comment by DionZac** [comment]
Probably obvious answer but need to clarify. If I run the project in the same browser (e.g. Chrome) but different version of it, the visitorId will be different right?

**Reply by Finesse** [reply]
It may be different, but not necessary will

**Reply by Finesse** [reply]
It's so because different browser versions can have different browser API implementations. Browser APIs are used by FingerprintJS to collect data for the fingerprint. Fingerprint is a simple hash sum of the data produced by the browser APIs. Therefore, when a browser API implementation used by FingerprintJS changes, the result fingerprint changes too.

Also, different browser versions can have the same implementations of the browser APIs used by FingerprintJS. In this case the fingerprint will be the same.
